### DISCUSSION TITLE
RAG with Langchain4J: Session Resources

### DISCUSSION THREAD

**Discussion by multispark** [microsoft/RAG_Hack#38] [Announcements]
This is the resources and discussion thread for this live stream: https://aka.ms/raghack/langchain4j
It will be updated with links after the session.

Recording: https://www.youtube.com/watch?v=h8RITrfumTU

Please ask any follow-up questions here, and the speakers will see them.

**Comment by jdubois** [comment]
Hey everyone! The demo code, documentation and slides are all available at https://github.com/jdubois/jdubois-langchain4j-demo

**Comment by lordjoe** [comment]
Where would I go to read about ALL the steps to set up this demo?
Also can the code handle multiple sources and can it hit the web and find added sources before finding an answer

**Reply by jdubois** [reply]
Hi!
- The steps are in the README file at https://github.com/jdubois/jdubois-langchain4j-demo/blob/main/README.md -> if it's not enough, ping me, I'll improve it!
- The code can handle multiple LLMs because they are configured as Spring Beans, with 2 profiles: there's an "azure" and a "local" profile, and depending on that profile it triggers either Azure OpenAI or Ollama.
- For hitting the Web, the code is at https://github.com/jdubois/jdubois-langchain4j-demo/blob/3f7c21fc4e149a170b9c9c9aab1285e1e9884b4e/src/main/java/com/example/demo/DemoController.java#L175 and that's one of the many utilities provided by LangChain4J, and that's why it's a good reason to use it :-)

**Comment by lordjoe** [comment]
how do I get to discord discussions

**Reply by multispark** [reply]
If you are looking for discord discussions, we do have a channel in the AI Discord:
[aka.ms/AzureAI/Discord](https://aka.ms/AzureAI/Discord)

Otherwise, we are using GitHub discussions as our main form of q&a!
